src/dashboard-refactor/search-results/types.ts

```diff
@@ -0,0 +1,143 @@
+import { TaskState } from 'ui-logic-core/lib/types'
+import { UIEvent } from 'ui-logic-core'
+
+import { AnnotationsSorter } from 'src/sidebar/annotations-sidebar/sorting'
+import {
+    AnnotationsSearchResponse,
+    StandardSearchResponse,
+} from 'src/search/background/types'
+
+export interface InteractionProps {
+    onCopyPasterBtnClick: React.MouseEventHandler
+    onListPickerBtnClick: React.MouseEventHandler
+    onTagPickerBtnClick: React.MouseEventHandler
+    onBookmarkBtnClick: React.MouseEventHandler
+    onNotesBtnClick: React.MouseEventHandler
+    onReplyBtnClick: React.MouseEventHandler
+    onShareBtnClick: React.MouseEventHandler
+    onTrashBtnClick: React.MouseEventHandler
+}
+
+export type PageInteractionProps = {
+    [Key in keyof InteractionProps]: (
+        day: number,
+        pageId: string,
+    ) => InteractionProps[Key]
+}
+
+export type NoteInteractionProps = {
+    [Key in keyof InteractionProps]: (noteId: string) => InteractionProps[Key]
+}
+
+export type SearchResultToState = (
+    result: AnnotationsSearchResponse | StandardSearchResponse,
+) => Pick<RootState, 'results' | 'noteData' | 'pageData'>
+export type SearchType = 'pages' | 'notes'
+export type NotesType = 'search' | 'user' | 'followed'
+
+export interface NormalizedState<T> {
+    allIds: string[]
+    byId: { [id: string]: T }
+}
+
+export interface NoteFormState {
+    isTagPickerShown: boolean
+    inputValue: string
+    tags: string[]
+}
+
+export interface NoteData {
+    url: string
+    tags: string[]
+    comment?: string
+    highlight?: string
+    displayTime: number
+}
+
+export interface PageData {
```

Comment:
Great suggestion. Will do.

> I'd like to put some work into our types at some point.

Yes, I would too! Particularly for fundamental data types to Memex that like pages and annotations that are used all over the place. I often have to write some code that deals with what (in my head) is just another page or annotation, but in that context I often cannot simply re-use an existing type as it might slightly vary in available fields or even the names of fields in that context, due to the way some code we need to interface with was written 2+years ago (or even way more recently). And also desiring not to go down the contextual rabbit hole of having to change said code, as it's often not directly related to what's being worked and might effect N other modules which I then also need to worry about.